### PR TITLE
Issue 240: Implement Boilerplate-Endpoint (5) 

### PR DESCRIPTION
**Pull request by DavidBrien**
### Definition of done:
- [x] The feature works -> production build
- [x] Tests exist and run successfully
- [x] All Tests work
- [x] Happy Path covered
- [x] No new Bugs/Followups

- [x] Documentation:
  - Readme.md
  - Doc-Section in PIT
  - Groovy/OpenAPI

---

### Release notes:
It's now possible to receive all boilerplates and specific boilerplates by id

---
## Code Review Checklist:
- [x] Does the code work?
- [x] Is all the code easily understood?
- [x] Are the function signatures named properly (function name, parameter names?)
- [x] Is the spelling correct?

- [x]  Is there redundant or duplicate code?
- [x]  Is there any commented out code?
- [x]  Can any logging or debugging code be removed?
- [x]  Is `for(i)` or `while` used? If so, remove
- [x]  Can any of the code be replaced with library functions?

-  Do you check input parameters (such as route parameters) for:
- [x] Type
- [x] Length
- [x] Format
- [x] Range?

- [x] Can comments be removed?
- [x] Is "external functionality" (API, Readme.md etc) documented/updated?
- [x] Is any unusual behavior or edge case handling described?

- [x] Do tests exist?
- [x] Do the snapshots look correct?
- [x] Are tests comprehensive?
- [x] Do unit tests actually test that the code is performing the intended functionality?

### CODE REVIEW

File: customizationArea/server/serviceMethods/pimApi.groovy [hunk 1]
```diff
@@ -85,6 +85,8 @@ class PitGroovyApi {
     private Closure generalDocumentPath =  { -> "/api/document"}
     private Closure documentPath = {documentId -> "${generalDocumentPath()}/$documentId"}
 
+    private Closure generalBoilerplatesPath = { -> "/api/boilerplates"}
```
Comment (UweStolz) [inline customizationArea/server/serviceMethods/pimApi.groovy]: Isn't it `boilerplate` instead of plural?

File: customizationArea/server/serviceMethods/pimApi.groovy [hunk 2]
```diff
@@ -758,6 +760,40 @@ class PitGroovyApi {
...
+     * @throws PIMUnreachableException
+     * @throws PIMInternalErrorException
+     */
+    public Response getAllBoilerplates() {
+        String path = generalBoilerplatesPath()
+        restGet(path)
+    }
+
+    /**
+     * Retrieve an boilerplates
```
Comment (UweStolz) [inline customizationArea/server/serviceMethods/pimApi.groovy]: Typo

File: customizationArea/server/serviceMethods/pimApi.groovy (hunk 1 above)
Comment (DavidBrien) [reply customizationArea/server/serviceMethods/pimApi.groovy]: this isn't wrong because it refers to the endpoint which is called boilerplates

File: customizationArea/server/serviceMethods/pimApi.groovy [hunk 3]
```diff
@@ -778,8 +778,8 @@ class PitGroovyApi {
     }
 
     /**
-     * Retrieve an boilerplates
-     * @param  boilerplatesId boilerplatesId
+     * Retrieve an boilerplate
```
Comment (UweStolz) [inline customizationArea/server/serviceMethods/pimApi.groovy]: It is `a` not `an`

File: customizationArea/server/serviceMethods/pimApi.groovy (hunk 1 above)
Comment (alexejFroebel) [reply customizationArea/server/serviceMethods/pimApi.groovy]: It is wrong

Comment (alexejFroebel) [reply customizationArea/server/serviceMethods/pimApi.groovy]: use singular

File: customizationArea/server/serviceMethods/pimApi.groovy [hunk 4]
```diff
@@ -85,6 +85,8 @@ class PitGroovyApi {
     private Closure generalDocumentPath =  { -> "/api/document"}
     private Closure documentPath = {documentId -> "${generalDocumentPath()}/$documentId"}
 
+    private Closure generalBoilerplatesPath = { -> "/api/boilerplates"}
+    private Closure boilerplatesPath = {boilerplatesId -> "${generalBoilerplatesPath()}/$boilerplatesId"}
```
Comment (alexejFroebel) [inline customizationArea/server/serviceMethods/pimApi.groovy]: Same here

File: customizationArea/server/serviceMethods/pimApi.groovy [hunk 5]
```diff
@@ -912,6 +914,44 @@ class PitGroovyApi {
...
+     * @throws NotAuthorizedException
+     * @throws UnknownHostException
+     * @throws GroovyAPIInternalErrorException
+     * @throws PITInternalErrorException
+     * @throws PIMAccessDeniedException
+     * @throws PIMUnreachableException
+     * @throws PIMInternalErrorException
+     */
+
+    public Response getBoilerplates(String boilerplateId) {
```
Comment (alexejFroebel) [inline customizationArea/server/serviceMethods/pimApi.groovy]: Singular
